prestoadmin/server.py

```diff
@@ -82,11 +86,202 @@ def new_sysnode_processor(node_info_rows):
 CONNECTOR_INFO_SQL = 'select catalog_name from system.metadata.catalogs'
 _LOGGER = logging.getLogger(__name__)
 
+DOWNLOAD_DIRECTORY = main_dir
+DEFAULT_RPM_NAME = 'presto-server-rpm.rpm'
+
+
+def _find_or_download_most_recent_presto_rpm():
+    latest_rpm_url = 'https://repository.sonatype.org/service/local/artifact/maven' \
+                     '/content?r=central-proxy&g=com.facebook.presto' \
+                     '&a=presto-server-rpm&e=rpm&v=RELEASE'
+    return _find_or_download_rpm(latest_rpm_url)
+
+
+def _check_if_valid_version(rpm_version):
+    """
+    This function returns whether the input rpm_version is valid.
+
+    The version is considered valid if it follows the format of
+    Semantic Versioning. That means it should have the form
+    X.Y.Z for Major.Minor.Patch. Z may not be applicable for
+    all rpm versions, so X.Y is also valid. X.Y.Z must all be numbers.
+
+    The version may be optionally followed by t (for teradata)
```

Comment:
I think appending f or t onto the version name to indicate where to download from is a bit hack-y given that it's never the actual last letter in a fb version, and is no longer the last letter in teradata versions.  I would rather have an optional argument to the command to indicate where to download from.